sam/record.go

```diff
@@ -173,6 +173,25 @@ func (r *Record) Strand() int8 {
 	return 1
 }
 
+// LessByName returns true if the receiver sorts by record name before other.
+func (r *Record) LessByName(other *Record) bool {
+	return r.Name < other.Name
+}
+
+// LessByCoordinate returns true if the receiver sorts by coordinate before other
+// according to the SAM specification.
+func (r *Record) LessByCoordinate(other *Record) bool {
+	rRefName := r.Ref.Name()
```

Comment:
The SAM spec says that coordinate sort is based on the name of the reference. The reason for this is that that value is stable while the ref id depends on the order of references in the header.